File: easy/shuffle_array/shuffle_array.go
package main

import "fmt"

func shuffle(nums []int, n int) []int {
	shuffledArray := make([]int, 0, 2*n)

	for i := 0; i < n; i++ {
		shuffledArray = append(shuffledArray, nums[i])
		shuffledArray = append(shuffledArray, nums[i+n])
	}

	return shuffledArray
}

func main() {
	fmt.Println("Solution 1 : ", shuffle([]int{2, 5, 1, 3, 4, 7}, 3))
	fmt.Println("Solution 2 : ", shuffle([]int{1, 2, 3, 4, 4, 3, 2, 1}, 4))
	fmt.Println("Solution 3 : ", shuffle([]int{1, 1, 2, 2}, 2))
}

File: easy/array/shuffle_array/shuffle_array.py
class Solution(object):
    def shuffle(self, nums, n):
        """
        :type nums: List[int]
        :type n: int
        :rtype: List[int]
        """

        shuffled_array = []
        for i in range(0, n):
            shuffled_array.append(nums[i])
            shuffled_array.append(nums[i+n])

        return shuffled_array


s = Solution()
print("Solution 1 :", s.shuffle([2, 5, 1, 3, 4, 7], 3))
print("Solution 2 :", s.shuffle([1, 2, 3, 4, 4, 3, 2, 1], 4))
print("Solution 1 :", s.shuffle([1, 1, 2, 2], 2))
